core/drivers/crypto/se050/core/rsa.c

```diff
@@ -15,6 +15,12 @@
 #include <tee/tee_cryp_utl.h>
 #include <tee_api_defines_extensions.h>
 
+static inline uint32_t oaep_alg_from_hash_alg(uint32_t hash_alg)
+{
+	return (TEE_ALG_RSAES_PKCS1_OAEP_MGF1_SHA1 & ~GENMASK_32(21, 20)) |
```

Comment:
I expect that it's bits [23:20] which are to replaced by bits [3:0] from `hash_alg`.
Please update the code accordingly and add a comment on what's going on.

Reply:
An explicit switch/case for the supported hash algo would maybe be easier to maintain.

Reply:
> An explicit switch/case for the supported hash algo would maybe be easier to maintain.

I dont see why....I'd rather handle the algorithm rather than the hash algorithm. Seems cleaner to me.